ui/src/pages/version/components/forms/components/transformer/OutputPanel.js

```diff
@@ -0,0 +1,269 @@
+import {
+  EuiDragDropContext,
+  euiDragDropReorder,
+  EuiDraggable,
+  EuiDroppable,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer
+} from "@elastic/eui";
+import { get, useOnChangeHandler } from "@gojek/mlp-ui";
+import { Panel } from "../Panel";
+import { AddButton } from "./components/AddButton";
+import { JsonOutputFieldCard } from "./components/table_outpus/JsonOutputFieldCard";
+import { BaseJsonOutputCard } from "./components/table_outpus/BaseJsonOutputCard";
+import {
+  BaseJson,
+  JsonOutput
+} from "../../../../../../services/transformer/TransformerConfig";
+
+const expandFields = flattenField => {
+  let temp = [];
+
+  flattenField.forEach(f => {
+    if (f.fieldName === undefined) {
+      return;
+    }
+
+    const nameSegments = splitName(f.fieldName);
+    let newField = {
+      ...f,
+      fieldName: nameSegments[nameSegments.length - 1]
+    };
+
+    temp = setValue(temp, nameSegments, newField);
+  });
+
+  return temp;
+};
+
+const splitName = name => {
+  // https://stackoverflow.com/questions/171480/regex-grabbing-values-between-quotation-marks
+  return (
+    name
+      // eslint-disable-next-line no-useless-escape
+      .split(/\.(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)/g)
+      // eslint-disable-next-line no-useless-escape
+      .map(n => n.replace(/\"/g, ""))
+  );
+};
+
+const setValue = (fields, nameSegment, fieldValue) => {
+  if (nameSegment.length <= 1) {
+    fields.push({ ...fieldValue, fields: [] });
+    return fields;
+  }
+
+  let field = searchField(fields, nameSegment[0]);
+  if (field === undefined) {
+    field = {
+      fieldName: nameSegment[0],
+      fields: []
+    };
+    fields.push(field);
+  }
+
+  setValue(field.fields, nameSegment.slice(1), fieldValue);
+  return fields;
+};
+
+const searchField = (fields, fieldName) => {
+  return fields && fields.find(f => f.fieldName === fieldName);
+};
+
+const flattenField = fields => {
+  let all = [];
+  fields.forEach(f => {
+    const flattenedFields = flatten(f, [f.fieldName]);
+    all = all.concat(flattenedFields);
+  });
+  return all;
+};
+
+const flatten = (field, path) => {
+  if (field.fields === undefined || field.fields.length === 0) {
+    return [
+      {
+        ...field,
+        fieldName: mergePath(path)
+      }
+    ];
+  }
+
+  let fields = [];
+  field.fields.forEach(f => {
+    path.push(f.fieldName);
+    fields = fields.concat(flatten(f, path));
+    path.pop();
+  });
+  return fields;
+};
+
+const mergePath = path => {
+  if (path.length === 1) {
+    return path[0];
+  }
+
+  return '"' + path.join('"."') + '"';
+};
+
+export const OutputPanel = ({ outputs = [], onChangeHandler, errors = {} }) => {
+  const { onChange } = useOnChangeHandler(onChangeHandler);
+
+  const [flattenedFields, setFlattenedFields] = useState([]);
+  useEffect(
+    () => {
+      const newFlattenedFields = flattenField(
+        outputs.length > 0 ? outputs[0].jsonOutput.jsonTemplate.fields : []
```

Comment:
hmmm i think we need to use last index of outputs instead hardcoded into 0? it is to handle if user is specifying multiple output, when this case happened output that will be used is the last output instead of first output